### PR TITLE
refactor(module:tree-select): refactor tree-select component

### PR DESCRIPTION
**Pull request by HsuanXyz**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/NG-ZORRO/ng-zorro-antd/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[x] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Application (the showcase website) / infrastructure changes
[ ] Other... Please describe:
```

## What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

### CODE REVIEW

File: components/tree/nz-tree.component.ts [hunk 1]
```diff
@@ -43,6 +61,9 @@ export class NzTreeComponent implements OnInit, OnChanges, OnDestroy {
   @Input() @InputBoolean() nzMultiple = false;
   @Input() @InputBoolean() nzExpandAll: boolean = false;
   @Input() @InputBoolean() nzHideUnMatched = false;
+  @Input() @InputBoolean() nzNoopAnimations = false;
```
Comment (vthinkxie) [inline components/tree/nz-tree.component.ts]: nzNoopAnimations -> nzNoAnimation

File: components/tree-select/nz-tree-select.component.ts [hunk 2]
```diff
@@ -192,7 +176,7 @@ export class NzTreeSelectComponent implements ControlValueAccessor, OnInit, Afte
 
   ngOnChanges(changes: SimpleChanges): void {
     if (changes.hasOwnProperty('nzNodes')) {
-      setTimeout(() => this.updateSelectedNodes(), 0);
+      setTimeout(() => this.updateSelectedNodes(true), 0);
```
Comment (vthinkxie) [inline components/tree-select/nz-tree-select.component.ts]: check this settimeout

File: components/tree-select/nz-tree-select.component.ts [hunk 3]
```diff
@@ -87,12 +84,12 @@ export class NzTreeSelectComponent implements ControlValueAccessor, OnInit, Afte
...
   @Input() @InputBoolean() nzDefaultExpandAll = false;
   @Input() nzNotFoundContent: string;
-  @Input() nzNodes: NzTreeNode[] = [];
+  @Input() nzNodes: Array<NzTreeNode | NzTreeNodeOptions> = [];
   @Input() nzOpen = false;
   @Input() nzSize = 'default';
   @Input() nzPlaceHolder = '';
   @Input() nzDropdownStyle: { [ key: string ]: string; };
-  @Input() nzDefaultExpandedKeys: string[] = [];
+  @Input() nzDefaultExpandedKeys: string[];
```
Comment (vthinkxie) [inline components/tree-select/nz-tree-select.component.ts]: do change this.

Comment (HsuanXyz) [reply components/tree-select/nz-tree-select.component.ts]: done

File: components/tree-select/nz-tree-select.component.html [hunk 4]
```diff
@@ -11,12 +11,28 @@
...
-       [ngStyle]="nzDropdownStyle">
+<ng-template
+  cdkConnectedOverlay
+  nzConnectedOverlay
+  [cdkConnectedOverlayOrigin]="cdkOverlayOrigin"
+  [cdkConnectedOverlayOpen]="nzOpen"
+  [cdkConnectedOverlayHasBackdrop]="true"
+  [cdkConnectedOverlayMinWidth]="nzDropdownMatchSelectWidth? null : triggerWidth"
+  [cdkConnectedOverlayWidth]="nzDropdownMatchSelectWidth? triggerWidth : null"
+  (backdropClick)="closeDropDown()"
```
Comment (vthinkxie) [inline components/tree-select/nz-tree-select.component.html]: miss (positionChange) here

**Pull request merged**